src/decorators/authorize.ts

```diff
@@ -3,61 +3,72 @@ import { AttributeValueMatch, AttributeValue } from "../types";
 
 import decorateWith from "./decorator";
 import OperationProcessor from "../processors/operation-processor";
-import { Resource } from "..";
+import Resource from "../resource";
+import ApplicationInstance from "../application-instance";
 
 type PrimitiveValue = string | number | boolean | object;
 
 const match = (actual: AttributeValue) => (item: PrimitiveValue) => (actual as any[]).includes(item);
+const ACCESS_RULES = Symbol("accessRules");
 
-function authorizeMiddleware(operation: Function, conditions: AttributeValueMatch[]) {
-  return function(this: OperationProcessor<Resource>) {
-    if (!this.appInstance.user) {
-      throw JsonApiErrors.Unauthorized();
-    }
+function conditionsPass(
+  appInstance: ApplicationInstance,
+  { attribute, value, operator = "some" }: AttributeValueMatch
+) {
+  const actual: AttributeValue = appInstance.user.attributes[attribute];
+  const expected = value;
```

Comment:
this could be done directly by destructuring :-)

Reply:
(I'm talking about the expected constant)

Reply:
Yes, but the typed syntax looks awful:

```ts
const [actual]: [actual: AttributeValue] = appInstance.user.attributes;
```

Reply:
```{ attribute:expected, value, operator = "some" }: AttributeValueMatch```

Reply:
This is actually wrong, we shouldn't destructure here. `attribute` is a dynamic name.